sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -390,6 +406,7 @@ private[sql] class DefaultWriterContainer(
 
   override def commitTask(): Unit = {
     try {
+      assert(writer != null, "OutputWriter instance should have been initialized")
```

Comment:
Should we throw an exception at here instead of using `assert`? Also, if `writer == null` is unexpected and we consider it as a bug, should we clearly let users know they are hitting a bug and they should report stack trace and information about the use case to us? I mean we need to be more specific on what a user needs to do when he/she gets `OutputWriter instance should have been initialized`.